pkg/clusteragent/admission/mutate/tags.go

```diff
@@ -37,6 +37,14 @@ var labelsToEnv = map[string]string{
 	kubernetes.VersionTagLabelKey: kubernetes.VersionTagEnvVar,
 }
 
+var labelsToWorkloadEnv = [][2]string{
+	{kubernetes.EnvTagLabelKey, "DD_WORKLOAD_ENV"},
+	{kubernetes.ServiceTagLabelKey, "DD_WORKLOAD_SERVICE"},
+	{kubernetes.KubeAppNameLabelKey, "DD_WORKLOAD_SERVICE"},
+	{kubernetes.VersionTagLabelKey, "DD_WORKLOAD_VERSION"},
+	{kubernetes.KubeAppVersionLabelKey, "DD_WORKLOAD_VERSION"},
```

Comment:
since some of them already correspond to the unified service tagging (DD_ENV, DD_SERVICE, DD_VERSION) why do we need to add new envvar?

Reply:
For env, you're right, I'll remove it.
For service and version, I added a fallback to `app.kubernetes.io/name` and `app.kubernetes.io/version` so that we are able to resolve the service without having the customer to explicitely specify one.